tests/client/amo/components/TestLink.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import {
+  findRenderedComponentWithType,
+  renderIntoDocument,
+} from 'react-addons-test-utils';
+import { Link } from 'react-router';
+
+import { LinkBase, mapStateToProps } from 'amo/components/Link';
+
+
+describe('<Link />', () => {
+  const defaultState = { api: { clientApp: 'android', lang: 'fr' } };
+
+  function render(props) {
+    return renderIntoDocument(
+      <LinkBase {...props}>Hi</LinkBase>);
+  }
+
+  it('uses clientApp and lang from API state', () => {
+    const props = mapStateToProps(defaultState);
+
+    assert(props.base, '/fr/android');
+  });
+
+  it('accepts an empty base', () => {
+    const root = render({ to: 'test' });
+
+    assert.equal(
+      findRenderedComponentWithType(root, Link).props.to, 'test');
+  });
+
+  it('allows overrides to base', () => {
+    const root = render({ base: '/en-US', to: '/categories' });
+
+    assert.equal(
+      findRenderedComponentWithType(root, Link).props.to, '/en-US/categories');
+  });
+
+  it('passes an object through as a to param', () => {
+    const root = render({ base: '/en-US', to: { pathname: '/categories' } });
+
+    assert.deepEqual(
+      findRenderedComponentWithType(root, Link).props.to,
+      { pathname: '/en-US/categories' }
+    );
+  });
+
+  it('passes other `to` types through to link', () => {
+    const root = render({ base: null, to: null });
+
+    assert.equal(root.props.base, null);
+    assert.deepEqual(root.props.to, null);
+  });
+
+  it('normalizes the path with a string', () => {
+    const root = render({ base: '/foo/', to: '/test' });
+
+    assert.equal(
+      findRenderedComponentWithType(root, Link).props.to, '/foo/test');
+  });
+
+  it('normalizes the path with an object', () => {
+    const root = render({ base: '/foo/', to: { pathname: '/test' } });
```

Comment:
I actually think the best API would just be to prepend everything and make it so the leading `/` in to `to` is meaningless but just a style thing, but I think that's ultimately bad.